File: app/src/main/java/com/reteno/sample/fragments/database/BaseReadViewHolder.java
package com.reteno.sample.fragments.database;

import androidx.recyclerview.widget.RecyclerView;
import androidx.viewbinding.ViewBinding;

abstract class BaseReadViewHolder<Model, ViewHolderBinding extends ViewBinding> extends RecyclerView.ViewHolder {
    final ViewHolderBinding bindingHolder;

    BaseReadViewHolder(ViewHolderBinding binding) {
        super(binding.getRoot());
        this.bindingHolder = binding;
    }


    protected abstract void bind(String timestamp, Model user);
}

File: app/src/main/java/com/reteno/sample/fragments/database/BaseReadViewHolder.kt
package com.reteno.sample.fragments.database

import androidx.recyclerview.widget.RecyclerView
import androidx.viewbinding.ViewBinding

internal abstract class BaseReadViewHolder<Model : Any, ViewHolderBinding : ViewBinding>(
    @JvmField val bindingHolder: ViewHolderBinding
) : RecyclerView.ViewHolder(bindingHolder.root) {
    abstract fun bind(model: Model)
}
